### ISSUE TITLE
mysqli_real_escape_string Cant fully prevent SQL Injection

### ISSUE THREAD

**Issue by zeroninefourfour** [mahtab2003/MOFHY-Lite#103]
**mysqli_real_escape_string** CAN PREVENT SQL Injection
BUT
it cant FULLY PREVENT SQL Injection.
So,Its Safe But Unsafe? xD
https://stackoverflow.com/questions/5741187/sql-injection-that-gets-around-mysql-real-escape-string

**Comment by zeroninefourfour** [comment]
Fo FIx It (Copied From Stackoverflow xD)
Use Modern Versions of MySQL (late 5.1, all 5.5, 5.6, etc) AND mysql_set_charset() / $mysqli->set_charset() / PDO's DSN charset parameter (in PHP ≥ 5.3.6)
Or
Don't use a vulnerable character set for connection encoding (you only use utf8 / latin1 / ascii / etc)

**Comment by mahtab2003** [comment]
Mofhy is expired

**Comment by greenreader9** [comment]
Wait, so you are no longer supporting it, _again_?

**Comment by ghost** [comment]
> Mofhy is expired

🤦

**Comment by jaikt** [comment]
🙄

**Comment by mahtab2003** [comment]
This time it is permanently closed and will no longer receive any updates from vendor

**Comment by jaikt** [comment]
why bro?

**Comment by mahtab2003** [comment]
There are some issues i can't resolve either.

**Comment by jaikt** [comment]
what are the issues? the community is there 🤔

**Comment by zeroninefourfour** [comment]
yep

**Comment by zeroninefourfour** [comment]
its easy to resolve :) just use PDO

**Comment by mahtab2003** [comment]
Xss and some tokenized system issue.
I'm thinking to make a new application based on codeigniter in order to make it work smoother and secure.

**Comment by greenreader9** [comment]
Go for it!

**Comment by jaikt** [comment]
> tokenized system issue

Like what? I would love to try fix it!

**Comment by greenreader9** [comment]
Well the client, and admin area login needs to be redone as well. Testing on my own install, both can be hacked. Not sharing details since, well, that should be obvious.

**Comment by mahtab2003** [comment]
> Well the client, and admin area login needs to be redone as well. Testing on my own install, both can be hacked. Not sharing details since, well, that should be obvious.

Yeah that's right and also logic bombs exists in this application

**Comment by zeroninefourfour** [comment]
wtf it just 1 day then lot of comment

**Comment by mahtab2003** [comment]
> wtf it just 1 day then lot of comment

So what?

**Comment by jaikt** [comment]
Can something like this being implemented correct? Reference:
https://youtu.be/I4JYwRIjX6c?t=996

**Comment by mahtab2003** [comment]
> Can something like this being implemented correct? Reference: https://youtu.be/I4JYwRIjX6c?t=996

Yes can be.

**Comment by jaikt** [comment]
Hey! I found a fix, it works on my local install. I tried with OOT and works! Will send a link to the new Repo since here it is no more maintained.

**Comment by jaikt** [comment]
I am making fix for prepared stmts. https://github.com/MofhyDevs/MofhyCommunity/

**Comment by greenreader9** [comment]
Um, there already is a community version...

**Comment by jaikt** [comment]
which no one cares to edit.

**Comment by mahtab2003** [comment]
> which no one cares to edit.

Well nobody cares whatever you are doing for community. They will just tell where you are wrong and will never try to fix it by there own.

**Comment by ghost** [comment]
> > which no one cares to edit.
> 
> Well nobody cares whatever you are doing for community. They will just tell where you are wrong and will never try to fix it by there own.

That's kinda sad honestly...

**Comment by jaikt** [comment]
> That's kinda sad honestly...

agreed

**Comment by greenreader9** [comment]
I mean you are not editing it either...

**Comment by mahtab2003** [comment]
> 

Who are you mentioning

**Comment by jaikt** [comment]
https://github.com/MyOwnFreeHostCommunity/Mofhy 🤔

**Comment by greenreader9** [comment]
I am talking to @jaikt 

Well the Community version here here: https://github.com/MOFH-Dev

You just made another one, when one already exists.

**Comment by mahtab2003** [comment]
> I am talking to @jaikt
> 
> Well the Community version here here: https://github.com/MOFH-Dev
> 
> You just made another one, when one already exists.

Well well well. That's a point

**Comment by Lebyy** [comment]
imo, rewriting the whole project from scratch but better this time is the best we can do rather than forking the same repo under a different organization everytime
OR
Create one organization where everyone who has already contributed / plans to contribute to the project with edit perms access under a private fork which will patch all security issues then release it after it's stable enough for production.

Ensuring all the security issues are patched this time.
